### PR TITLE
SAK-47987 grader. Add peer review information to submission

### PR DESCRIPTION
**Pull request by adrianfish**
https://sakaiproject.atlassian.net/browse/SAK-47987

### CODE REVIEW

**Comment by adrianfish** [comment]
Screenshot 2023-10-19 at 16 26 16

File: assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java [hunk 1]
```diff
@@ -763,6 +767,61 @@ private Map<String, Object> submissionToMap(Set<String> activeSubmitters, Assign
...
+                                    if (assignment.getIsGroup()) {
+                                        String siteId = toolManager.getCurrentPlacement().getContext();
+                                        Site site = siteService.getSite(siteId);
+                                        review.setAssessorDisplayName(site.getGroup(review.getId().getAssessorUserId()).getTitle());
+                                    } else {
+                                        review.setAssessorDisplayName(userDirectoryService.getUser(review.getId().getAssessorUserId()).getDisplayName());
+                                    }
+                                } catch (IdUnusedException | UserNotDefinedException e) {
+                                    //reviewer doesn't exist or one of userId/groupId/siteId is wrong
+                                    log.error(e.getMessage(), e);
```
Comment (ern) [inline assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java:796]:
```suggestion
                                    log.warn("Could not fetch site [{}]", siteId, e.toString());
```

File: assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java [hunk 2]
```diff
@@ -763,6 +767,61 @@ private Map<String, Object> submissionToMap(Set<String> activeSubmitters, Assign
...
+                        if (!review.getRemoved() && (review.getScore() != null || (StringUtils.isNotBlank(review.getComment())))) {
+                            //only show peer reviews that have either a score or a comment saved
+                            if (assignment.getPeerAssessmentAnonEval()) {
+                                //annonymous eval
+                                review.setAssessorDisplayName(rb.getFormattedMessage("gen.reviewer.countReview", completedReviews.size() + 1));
+                            } else {
+                                //need to set the assessor's display name
+                                try {
+                                    if (assignment.getIsGroup()) {
+                                        String siteId = toolManager.getCurrentPlacement().getContext();
```
Comment (ern) [inline assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java:788]: move this before the try

File: assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java [hunk 3]
```diff
@@ -763,6 +767,61 @@ private Map<String, Object> submissionToMap(Set<String> activeSubmitters, Assign
...
+                                        Site site = siteService.getSite(siteId);
+                                        review.setAssessorDisplayName(site.getGroup(review.getId().getAssessorUserId()).getTitle());
+                                    } else {
+                                        review.setAssessorDisplayName(userDirectoryService.getUser(review.getId().getAssessorUserId()).getDisplayName());
+                                    }
+                                } catch (IdUnusedException | UserNotDefinedException e) {
+                                    //reviewer doesn't exist or one of userId/groupId/siteId is wrong
+                                    log.warn("Could not fetch site [{}]", siteId, e.toString());
```
Comment (ern) [inline assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java:796]:
```suggestion
                                    log.warn("Could not fetch site [{}], {}", siteId, e.toString());
```

File: assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java [hunk 4]
```diff
@@ -763,6 +767,61 @@ private Map<String, Object> submissionToMap(Set<String> activeSubmitters, Assign
...
+                            // get attachments for peer review item
+                            List<PeerAssessmentAttachment> attachments = assignmentPeerAssessmentService.getPeerAssessmentAttachments(review.getId().getSubmissionId(), review.getId().getAssessorUserId());
+                            if (attachments != null && !attachments.isEmpty()) {
+                                List<Reference> attachmentRefList = new ArrayList<>();
+                                for (PeerAssessmentAttachment attachment : attachments) {
+                                    try {
+                                        Reference ref = entityManager.newReference(contentHostingService.getReference(attachment.getResourceId()));
+                                        attachmentRefList.add(ref);
+                                    } catch (Exception e) {
+                                        log.warn(e.getMessage(), e);
```
Comment (ern) [inline assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java:810]: better log message

**Review by ern — APPROVED** [review]

**Pull request merged**